src/modules/agora/AgoraClient.ts

```diff
@@ -0,0 +1,131 @@
+/*******************************************************************************
+
+    The class that recover data
+
+    Copyright:
+        Copyright (c) 2020 BOS Platform Foundation Korea
+        All rights reserved.
+
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+import axios, { AxiosInstance, AxiosResponse, AxiosError } from 'axios';
+import URI from "urijs";
+
+/**
+ * The class that recover data
+ */
+export class AgoraClient
+{
+    /**
+     * The network address to connect to the Agora
+     */
+    private host: string;
+
+    /**
+     * The network port to connect to the Agora
+     */
+    private port: string;
+
+    /**
+     * The instance of the axios
+     */
+    private client: AxiosInstance;
+
+    /**
+     * Constructor
+     * @param host - The network address to connect to the Agora
+     * @param port - The network port to connect to the Agora
+     */
+    constructor (host: string, port: string)
+    {
+        this.client = axios.create();
```

Comment:
```suggestion
        this.client = axios.create({
            baseURL: host + port
        });
```
Don't it need baseURL basic settings for `Axios.create`?

Reply:
`this.client.defaults.timeout = 3000;`

It can also set a timeout.

Reply:
Thank you for your good suggestion.
I changed it as below.

```
    /**
     * The URL encoder
     */
    private uri: URI;

    /**
     * Constructor
     * @param host - The network address to connect to the Agora
     * @param port - The network port to connect to the Agora
     */
    constructor (host: string, port: string)
    {
        this.client = axios.create();
        this.client.defaults.timeout = 10000;
        this.uri = URI(host).port(port);
    }
```